src/sql/workbench/parts/notebook/browser/cellViews/collapse.component.ts

```diff
@@ -0,0 +1,81 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import 'vs/css!./code';
+
+import { OnInit, Component, Input, Inject, forwardRef, ElementRef, ChangeDetectorRef, ViewChild, SimpleChange, OnChanges } from '@angular/core';
+import { CellView } from 'sql/workbench/parts/notebook/browser/cellViews/interfaces';
+import { ICellModel } from 'sql/workbench/parts/notebook/browser/models/modelInterfaces';
+
+export const COLLAPSE_SELECTOR: string = 'collapse-component';
+
+@Component({
+	selector: COLLAPSE_SELECTOR,
+	templateUrl: decodeURI(require.toUrl('./collapse.component.html'))
+})
+
+export class CollapseComponent extends CellView implements OnInit, OnChanges {
+	@ViewChild('collapseCellButton', { read: ElementRef }) private collapseCellButtonElement: ElementRef;
+	@ViewChild('expandCellButton', { read: ElementRef }) private expandCellButtonElement: ElementRef;
+
+	@Input() cellModel: ICellModel;
+	@Input() activeCellId: string;
+
+	constructor(
+		@Inject(forwardRef(() => ChangeDetectorRef)) private _changeRef: ChangeDetectorRef,
+	) {
+		super();
+	}
+
+	ngOnInit() {
+	}
+
+	ngOnChanges(changes: { [propKey: string]: SimpleChange }) {
+	}
+
+	ngAfterContentInit() {
+		this._register(this.cellModel.onCollapseStateChanged(isCollapsed => {
+			this.onCellCollapse(isCollapsed);
+		}));
+		this.onCellCollapse(this.cellModel.isCollapsed);
+		if (this.activeCellId === this.cellModel.id) {
+			this.toggleIconVisibility(true);
+		}
+	}
+
+	private onCellCollapse(isCollapsed: boolean): void {
```

Comment:
nit: i was confused for a second that this was a method and not an event, as it starts with "on".

Reply:
Would "handleCellCollapse" be better here?

Reply:
sounds great!